Mathlib/CategoryTheory/ConnectedComponents.lean

```diff
@@ -40,6 +39,32 @@ def ConnectedComponents (J : Type u₁) [Category.{v₁} J] : Type u₁ :=
 instance [Inhabited J] : Inhabited (ConnectedComponents J) :=
   ⟨Quotient.mk'' default⟩
 
+/-- Every function from connected components of a category gives a functor to discrete category -/
+def ConnectedComponents.functorToDiscrete {X : Type*}
+    (f : ConnectedComponents J → X) : J ⥤ Discrete X where
+  obj Y :=  Discrete.mk (f (Quotient.mk (Zigzag.setoid _) Y))
+  map g := Discrete.eqToHom (congrArg f (Quotient.sound (Zigzag.of_hom g)))
+
+/-- Every functor to a discrete category gives a function from connected components -/
+def ConnectedComponents.liftFunctor {X : Type*} (F :J ⥤ Discrete X) :
+    (ConnectedComponents J → X) :=
+  Quotient.lift (fun c => (F.obj c).as)
+    (fun _ _ h => eq_of_zigzag X (zigzag_obj_of_zigzag F h))
+
+/-- Functions from connected components and functors to discrete category are in bijection -/
+def ConnectedComponents.typeToCatHomEquiv   (X : Type*) :
```

Comment:
```suggestion
def ConnectedComponents.typeToCatHomEquiv (X : Type*) :
```

Reply:
Also, I think you want `J` to be explicit here?

Reply:
how do you see that ?

Reply:
Because there is no way to infer it from other variables

Reply:
Or maybe implicit in what we have, explicit in what we need

Reply:
> It might make sense to be explicit elsewhere too

Not in the toFun and invFun, because there it can be inferred from the argument